### DISCUSSION TITLE
[Feature Request] Sorting within groups

### DISCUSSION THREAD

**Discussion by fluffy-critter** [TeamRizu/OutFox#240] [03 - Feature Requests]
**Is your feature request related to a problem? Please describe.**

I have a bunch of song packs installed, and I'd like to be able to sort my songs within those song packs by e.g. difficulty without mixing the song packs together.

**Describe the solution you'd like**

Either some means of specifying sorting and grouping separately from the music wheel interface, or a global preference for "keep original groups." It would also be fine if this could be done with a simple theme mod (and if that mod were documented).

**Additional context**

I have 1650 songs in 24 packs, each which have their own weird arbitrary difficulty scaling. I've only just started using Stepmania and I'm already overwhelmed with selecting what to play next.

**Comment by ListenerJubatus** [comment]
We're looking into it for a future release.

**Comment by ListenerJubatus** [comment]
@fluffy-critter this mockup shows how such an interface would look in the Alpha 5.0.0 theme. Is this what you had in mind?
![ScreenSelectMusic song](https://user-images.githubusercontent.com/6346433/108918073-5f86b380-75f6-11eb-8ded-932164255dd5.png)

**Comment by fluffy-critter** [comment]
Hm, maybe. I was thinking more that it would maintain the same song wheel UI, it's just that the order of the songs within the wheel would be changed such that it's still grouped by the original groups but the order within the group is based on the sort criterion (difficulty, BPM, title, etc.), and maybe taking the opportunity to add an optional in-group order field so that songpack creators can provide a default ordering for things instead of being in title order.

**Comment by ListenerJubatus** [comment]
@fluffy-critter that's what I envisioned. I play Arcaea and that basically works the same way.

**Comment by fluffy-critter** [comment]
I guess I'm mostly not clear on how this mockup maps to the standard Stepmania UX, especially when it's being run from a dance pad alone. What I was thinking would be maybe a thing that gets added to the sort options popup (which I forget how it gets triggered on the base outfox theme but is by pressing left and right simultaneously in Simply Love). I've only been playing with Outfox and Alpha 4.9.something and I'm not even clear on how to get the Alpha 5.0.0 theme right now (but I haven't checked the public outfox builds in a while either).

**Comment by ListenerJubatus** [comment]
@fluffy-critter the new default will support all input modes, but touch/mouse support is a major priority.
